.infrastructure/helm-chart/todoapp/templates/deployment.yml

```diff
@@ -0,0 +1,86 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{.Chart.Name}}
+  namespace: {{.Values.todoapp.namespace}}
+spec:
+  strategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxSurge: {{.Values.todoapp.rollingUpdate.maxSurge}}
+      maxUnavailable: {{.Values.todoapp.rollingUpdate.maxUnavailable}}
```

Comment:
Ensure that the `todoapp.rollingUpdate.maxSurge` and `todoapp.rollingUpdate.maxUnavailable` values are defined in your `values.yaml` file. If they are not defined, this could lead to issues when deploying the chart. Consider providing default values or handling the case where they might be undefined.